src/rebar_raw_resource.erl

```diff
@@ -348,16 +476,19 @@ format_error(Mod, Func, Class, Data) ->
 absorb_state(State) ->
     Data1 = mod_data(),
     Data2 = absorb_resources(Data1, rebar_state:resources(State)),
-    Data3 = absorb_deps(Data2, rebar_state:get(State, 'deps', [])),
-    Data4 = absorb_app_infos(Data3, rebar_state:lock(State)),
+    Data3 = absorb_app_infos(Data2, rebar_state:lock(State)),
+    Data4 = absorb_deps(Data3, rebar_state:get(State, 'deps', [])),
 %   Data5 = absorb_profiles(Data4, rebar_state:current_profiles(State), State),
     mod_data(dump_mod_data(Data4)).
 
 -spec absorb_app_infos(Data :: mod_data(), AppInfos :: [tuple()])
         -> mod_data() | no_return().
 absorb_app_infos(Data, [AppInfo | AppInfos]) ->
+    % absorb deps from app_info to ensure that all deps are absorbed
+    AppInfoDeps = rebar_app_info:get(AppInfo, deps, []),
```

Comment:
This is another key call-out. I found that when there is a "raw" dependency of a dependency in a top-level application, it's often not included in the "deps" or "lock" rebar state. This can trigger an `unmapped_dep_loc` or `unmapped_dep_name` error when downloading a missing raw dep because of the rebar state. This call collects dependencies of dependencies, which appears to resolve the problem. If there is a more effective way to reliablty collect all the app specifications, that would be nice.

Reply:
I think you should add a pointer to this PR somewhere in the comments for this function.